samtools.wdl

```diff
@@ -136,34 +176,50 @@ task Merge {
     }
 }
 
-task SortByName {
+task Sort {
     input {
-        File bamFile
-        String outputBamPath = "namesorted.bam"
+        File inputBam
+        String outputPrefix
+        Boolean sortByName = false
+        String outputFormat = "BAM"
 
-        String dockerImage = "quay.io/biocontainers/samtools:1.8--h46bd0b3_5"
+        Int cores = 1
+        String memory = "2G"
+        String dockerImage = "quay.io/biocontainers/samtools:1.10--h9402c20_2"
     }
 
     command {
         set -e
-        mkdir -p "$(dirname ~{outputBamPath})"
-        samtools sort -n ~{bamFile} -o ~{outputBamPath}
+        mkdir -p "$(dirname ~{outputPrefix})"
+        samtools sort \
+        ~{true="-n" false="" sortByName} \
+        "--output-fmt " ~{outputFormat} \
+        --threads ~{cores} \
```

Comment:
`~{"--threads " + threads}` If no additional cores are given this flag should not be used.